### ISSUE TITLE
[v5] undefined state is being replaced by empty object in persistReducer

### ISSUE THREAD

**Issue by chmanie** [rt2zz/redux-persist#429]
In my reducers I am using the pattern

```javascript
export const reducer = (state = INITIAL_STATE, action) => {
 // reducer code
};
```

to assign a default value if the state is not given (in this case the `INITIAL_STATE`). This will also happen with any other ES5 test whether the state is defined.

When wrapping this reducer using `persistReducer`, the state is _always_ defined due to these lines:

https://github.com/rt2zz/redux-persist/blob/v5/src/persistReducer.js#L51-L52

That means the default value (when calling the reducer for the first time by `@@INIT`) will never be applied hence, I loose the ability to define my default state like this. I think there should be a guard checking for an undefined initial state and just pass this value to the baseReducer (along those lines):

```javascript
let restState: State
if (!state) {
  restState = state // use initial value of undefined here
} else {
  let { _persist, ...rest } = state || {} // I think this just strips the _persist, hence the local var should be fine
  restState = rest
}
```

If I'm misunderstanding this or you want it to be handled differently, let me know. I'll happily provide a PR for this as well.

v4 handled that differently (I didn't use it but checked the code) by just passing the state to the reducer without modifying it. So the problem should not exist there.

**Comment by rt2zz** [comment]
ya good catch, this was recently broken while trying to optimize persistReducer.

I am thinking maybe the logic is actually to have a more strict passthrough case (at the end of persistReducer). This seems to make sense since all of these lines https://github.com/rt2zz/redux-persist/blob/v5/src/persistReducer.js#L116-L120 are irrelevant if PERSIST has not yet been handled.

```js
// if we have not already handled PERSIST or state is undefined, straight passthrough
if (!_persist || !state) return baseReducer(state, action)

// otherwise, pull off _persist, run the reducer, and update the persistoid
let newState = {
  ...baseReducer(restState, action),
  _persist,
}
_persistoid && _persistoid.update(newState)
return newState
```

**Comment by chmanie** [comment]
> I am thinking maybe the logic is actually to have a more strict passthrough case

I agree, my proposed solution was rather a quick fix for me. Let me know if you need help :)

**Comment by rt2zz** [comment]
fix is out in 5.0.0-beta.3 👍
